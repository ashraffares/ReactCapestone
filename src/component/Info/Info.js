/* eslint-disable camelcase */
import React from 'react';
import propTypes from 'prop-types';
import styles from './info.module.css';

const Info = ({ pokemon }) => {
  const {
    name, height, weight, base_experience, sprites, abilities, moves,
  } = pokemon;

  const {
    back_default, back_shiny, front_default, front_shiny,
  } = sprites;
  return (
    <section className={styles.info}>
      <div className={styles.imgContainer}>
        <img src={back_default} alt={name} />
        <img src={back_shiny} alt={name} />
        <img src={front_default} alt={name} />
        <img src={front_shiny} alt={name} />
      </div>
      <div className={styles.basicInfo}>
        <h3 className={styles.headers}>BasicInfo: </h3>
        <ol type="I" className={styles.ul}>
          <li className={styles.li}>{`name: ${name}`}</li>
          <li className={styles.li}>{`height: ${height}`}</li>
          <li className={styles.li}>{`weight: ${weight}`}</li>
          <li className={styles.li}>{`base_experience: ${base_experience}`}</li>
        </ol>
      </div>
      <div className={styles.ability}>
        <h3 className={styles.headers}>Abilites: </h3>
        <ol type="I" className={styles.ul}>
          {
          abilities.map((el) => (
            <li className={styles.li} key={el.ability.name}>{el.ability.name}</li>))
          }
        </ol>
      </div>

      <div className={styles.moves}>
        <h3 className={styles.headers}>Moves: </h3>
        <ol type="I" className={styles.ul}>
          {
            moves.map((el) => <li className={styles.li} key={el.move.name}>{el.move.name}</li>)
          }
        </ol>
      </div>
    </section>
  );
};

Info.propTypes = {
  pokemon: propTypes.shape({
    name: propTypes.string.isRequired,
    height: propTypes.number.isRequired,
    weight: propTypes.number.isRequired,
    base_experience: propTypes.number.isRequired,
    sprites: propTypes.shape({
      back_default: propTypes.string.isRequired,
      back_shiny: propTypes.string.isRequired,
      front_default: propTypes.string.isRequired,
      front_shiny: propTypes.string.isRequired,
    }).isRequired,
    abilities: propTypes.arrayOf(
      propTypes.shape({
        ability: propTypes.shape({
          name: propTypes.string.isRequired,
        }).isRequired,
      }).isRequired,
    ),
    moves: propTypes.arrayOf(
      propTypes.shape({
        move: propTypes.shape({
          name: propTypes.string.isRequired,
        }).isRequired,
      }).isRequired,
    ).isRequired,
  }).isRequired,
};

export default Info;
