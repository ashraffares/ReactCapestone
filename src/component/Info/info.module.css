.info {
  max-width: 80vw;
  margin: 30px auto;
  align-items: center;
}

.headers {
  margin: 0;
  padding-left: 20px;
  font-size: 2vw;
}

.ul {
  width: fit-content;
  margin: 0;
}

.imgContainer {
  width: 100%;
  display: flex;
  justify-content: space-evenly;
  margin: 20px 0;
  padding: 20px 0;
  background-color: #fff;
  border: 1px solid #e8e8e8;
  border-radius: 8px;
  box-shadow: 4px 8px 6px #aaa;
}

.imgContainer img:hover {
  box-shadow: 4px 8px 6px turquoise;
  border: 1px solid #e8e8e8;
}

.basicInfo,
.ability,
.moves {
  max-width: inherit;
  display: inline-flex;
  width: 100%;
  margin: 20px 0;
  padding: 20px 0;
  background-color: #fff;
  border: 1px solid #e8e8e8;
  border-radius: 8px;
  box-shadow: 4px 8px 6px #aaa;
}
