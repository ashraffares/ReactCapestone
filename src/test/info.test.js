import renderer from 'react-test-renderer';
import Info from '../component/Info/Info';

it('it matches the snapshot', () => {
  const info = renderer.create(Info).toJSON();
  expect(info).toMatchSnapshot();
});
