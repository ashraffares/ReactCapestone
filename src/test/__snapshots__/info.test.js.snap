// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it matches the snapshot 1`] = `null`;
